From edd344986fc767354997b15383d909a66f86535e Mon Sep 17 00:00:00 2001
From: shengfei Xu <xsf@rock-chips.com>
Date: Fri, 9 Feb 2018 14:25:58 +0800
Subject: [PATCH] power: rk817: fix the error when plug in OTG

[   36.485506] [<ffffff80080b97c0>] __queue_delayed_work+0x6c/0x148
[   36.485523] [<ffffff80080b98ec>] queue_delayed_work_on+0x50/0x6c
[   36.485543] [<ffffff80087315c4>] rk817_charge_host_evt_notifier+0x20/0x2c
[   36.485561] [<ffffff80080c0c54>] notifier_call_chain+0x48/0x80
[   36.485576] [<ffffff80080c0cdc>] raw_notifier_call_chain+0x14/0x1c
[   36.485596] [<ffffff8008855b10>] extcon_sync+0x74/0x1c4
[   36.485615] [<ffffff80083c3118>] rockchip_usb2phy_id_irq+0x270/0x2b8
[   36.485633] [<ffffff80080f84ac>] irq_thread_fn+0x28/0x68
[   36.485648] [<ffffff80080f8070>] irq_thread+0x100/0x1d8
[   36.485662] [<ffffff80080bfc80>] kthread+0xe8/0xf0
[   36.485678] [<ffffff80080832a0>] ret_from_fork+0x10/0x30

Change-Id: Ib1a50dbf2210e43efb056835e4fef1caae2ca059
Signed-off-by: shengfei Xu <xsf@rock-chips.com>
---
 drivers/power/rk817_charger.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/power/rk817_charger.c b/drivers/power/rk817_charger.c
index c2c426e608f0..2378f5b449be 100644
--- a/drivers/power/rk817_charger.c
+++ b/drivers/power/rk817_charger.c
@@ -1083,9 +1083,9 @@ static int rk817_charge_host_evt_notifier(struct notifier_block *nb,
 					  unsigned long event, void *ptr)
 {
 	struct rk817_charger *charge =
-		container_of(nb, struct rk817_charger, cable_discnt_nb);
+		container_of(nb, struct rk817_charger, cable_host_nb);
 
-	queue_delayed_work(charge->usb_charger_wq, &charge->discnt_work,
+	queue_delayed_work(charge->usb_charger_wq, &charge->host_work,
 			   msecs_to_jiffies(10));
 
 	return NOTIFY_DONE;
-- 
2.35.3

